src/item_pocket.cpp

```diff
@@ -1504,17 +1503,17 @@ ret_val<item_pocket::contain_code> item_pocket::_can_contain( const item &it,
         }
     }
 
-    if( data->ablative && !contents.empty() ) {
-        if( contents.front().can_combine( it ) ) {
-            // Only items with charges can succeed here.
-            return ret_val<item_pocket::contain_code>::make_success();
-        } else {
-            return ret_val<item_pocket::contain_code>::make_failure(
-                       contain_code::ERR_NO_SPACE, _( "ablative pocket already contains a plate" ) );
+    if( data->ablative ) {
```

Comment:
Is there a reason to split one `if` into two?

Reply:
I guess the reason is to combine following check at the line 1517.
Transform this code
```C++
 if( data->ablative && !contents.empty() ) {
 }

 if( data->ablative ) {
    if( it.is_rigid() ) {
    }
 } 
```
into that:
```C++
 if( data->ablative ) {
    if ( !contents.empty() ) {
    }
    if( it.is_rigid() ) {
    }
 } 
```

Reply:
Yes, it's to compact the code a bit, but it's also because I found out the hard way that the section of code around the `!contents.empty` check must be called before the `is_rigid` section or else there's a risk of causing an infinite loop. Combining them into one large block makes it more clear the order should be kept like that.